buildpacks/ruby/src/layers/bundle_download_layer.rs

```diff
@@ -1,149 +1,137 @@
-            cmd.named_output().map_err(|error| {
-                fun_run::map_which_problem(error, cmd.mut_cmd(), self.env.get("PATH").cloned())
-            })
-        })
-        .map_err(RubyBuildpackError::GemInstallBundlerCommandError)?;
-
-        LayerResultBuilder::new(self.metadata.clone())
-            .env(
-                LayerEnv::new()
-                    .chainable_insert(Scope::All, ModificationBehavior::Delimiter, "PATH", ":")
-                    .chainable_insert(
-                        Scope::All,
-                        ModificationBehavior::Prepend,
-                        "PATH", // Ensure this path comes before default bundler that ships with ruby, don't rely on the lifecycle
-                        bin_dir,
+            invalid_metadata_action: &|_| {
+                (
+                    InvalidMetadataAction::DeleteLayer,
+                    "invalid metadata".to_string(),
+                )
+            },
+            restored_layer_action: &|old: &Metadata, _| {
+                if let Some(cause) = metadata_diff(old, &metadata) {
+                    (RestoredLayerAction::DeleteLayer, cause)
+                } else {
+                    (
+                        RestoredLayerAction::KeepLayer,
+                        "using cached version".to_string(),
                     )
-                    .chainable_insert(Scope::All, ModificationBehavior::Delimiter, "GEM_PATH", ":")
-                    .chainable_insert(
-                        Scope::All,
-                        ModificationBehavior::Prepend,
-                        "GEM_PATH", // Bundler is a gem too, allow it to be required
-                        gem_path,
-                    ),
-            )
-            .build()
-    }
-
-    fn existing_layer_strategy(
-        &mut self,
-        _context: &BuildContext<Self::Buildpack>,
-        layer_data: &LayerData<Self::Metadata>,
-    ) -> Result<ExistingLayerStrategy, RubyBuildpackError> {
-        let old = &layer_data.content_metadata.metadata;
-        let now = self.metadata.clone();
-        match cache_state(old.clone(), now) {
-            State::NothingChanged(_version) => {
-                log_step("Using cached version");
-
-                Ok(ExistingLayerStrategy::Keep)
+                }
+            },
+        },
+    )?;
+
+    let bin_dir = layer_ref.path().join("bin");
+    let gem_path = layer_ref.path();
+    match &layer_ref.state {
+        LayerState::Restored { cause: _ } => {
+            bullet = bullet.sub_bullet("Using cached version");
+        }
+        LayerState::Empty { cause } => {
+            match cause {
+                EmptyLayerCause::NewlyCreated => {}
+                EmptyLayerCause::InvalidMetadataAction { cause }
+                | EmptyLayerCause::RestoredLayerAction { cause } => {
+                    bullet = bullet.sub_bullet(format!("Clearing cache {cause}"));
```

Comment:
The log output was changed in this PR, to include more details when the bundler version changed, but that also made the log output a bit awkward:

    Clearing cache Bundler version (`2.4.5` to `2.4.17`)

The `cause` also has differing casing (`Bundler version` vs `invalid metadata`).

```suggestion
                    bullet = bullet.sub_bullet(format!("Clearing cache {}", style::details(cause)));
```

This (combined with the edit below, fixing the bundler version change `cause`) is one way this could be addressed. Might make sense to consider a different phrasing though as these changes result in:

    Clearing cache (bundler version changed (`2.4.5` to `2.4.17`))